File: Tarefa3/scene_graph/python/color.py
import glm
from appearance import Appearance

class Color(Appearance):
    def __init__ (self, r, g, b, a=1):
      self.color = glm.vec4(r,g,b,a)

    def Load (self, st):
      shd = st.GetShader()
      shd.SetUniform("color",self.color)

File: Tarefa_2.1/color.cpp
#include "color.h"
#include "shader.h"
#include "state.h"

#ifdef _WIN32
//#define GLAD_GL_IMPLEMENTATION // Necessary for headeronly version.
#include <glad/gl.h>
#elif __APPLE__
#include <OpenGL/gl3.h>
#else
#include <GL/glew.h>
#endif

ColorPtr Color::Make (float r, float g, float b, float a)
{
  return ColorPtr(new Color(r,g,b,a));
}

Color::Color (float r, float g, float b, float a)
: m_color(r,g,b,a)
{
}
Color::~Color ()
{
}
void Color::Load (StatePtr st)
{
  ShaderPtr shd = st->GetShader();
  shd->SetUniform("color",m_color);
}
